variables.tf

```
@@ -0,0 +1,40 @@
+variable "backend_config" {
+  description = "Object of S3 backend config"
+  type = object({
+    bucket        = string
+    force_destroy = optional(bool)
+    versioning    = optional(string)
+    public_access_block = optional(object({
+      block_public_acls       = bool
+      block_public_policy     = bool
+      ignore_public_acls      = bool
+      restrict_public_buckets = bool
+    }))
+    server_side_encryption_configuration = optional(object({
+      bucket_key_enabled = bool
+      sse_algorithm      = string
+      kms_master_key_id  = string
+    }))
+    dynamodb_table = optional(object({
+      name                        = string
+      deletion_protection_enabled = bool
+      billing_mode                = string
+    }))
```

Let's make the `dynamodb_table` object required, with a couple optional attributes of its own.

```suggestion
    dynamodb_table = object({
      name                        = string
      deletion_protection_enabled = optional(bool, true)
      billing_mode                = optional(string, "PAY_PER_REQUEST")
    })
```